flux-kube/plugins/flux-kube.c

```diff
@@ -0,0 +1,352 @@
+        goto done;
+    }
+    if (flux_jobtap_event_post_pack (p,
+                                     *idptr,
+                                     "alloc",
+                                     "{s:b}",
+                                     "bypass", true) < 0)
+        flux_jobtap_raise_exception (p,
+                                     *idptr,
+                                     "alloc", 0,
+                                     "failed to post alloc event: %s",
+                                     strerror (errno));
+
+    /*  Set "needs-free" so that flux-kube knows that a "free"
+     *  event needs to be emitted for this node.
+     */
+    if (flux_jobtap_job_aux_set (p, *idptr, "flux-kube::free", p, NULL) < 0)
+        flux_log_error (flux_jobtap_get_flux (p),
+                        "id=%ju: Failed to set flux-kube::free",
+                        (intmax_t)(*idptr));
+
+done:
+    flux_future_destroy (f);
+}
+
+static void get_response_continuation (flux_future_t *f, void *arg)
+{
+    flux_plugin_t *p = arg;
+
+    if (flux_rpc_get (f, NULL) != 0)
+        flux_log_error (flux_jobtap_get_flux (p),
+                        "Failed to unpack RPC from flux-kube.py");
+
+
+    flux_future_destroy (f);
+}
+
+static int alloc_start (flux_plugin_t *p,
+                        flux_jobid_t id,
+                        const char *R)
+{
+    flux_t *h;
+    char key[64];
+    flux_future_t *f = NULL;
+    flux_kvs_txn_t *txn = NULL;
+    flux_jobid_t *idptr = NULL;
+
+    if (!(h = flux_jobtap_get_flux (p))
+        || flux_job_kvs_key (key, sizeof (key), id, "R") < 0
+        || !(txn = flux_kvs_txn_create ())
+        || flux_kvs_txn_put (txn, 0, key, R) < 0
+        || !(f = flux_kvs_commit (h, NULL, 0, txn))
+        || flux_future_then (f, -1, alloc_continuation, p)
+        || !(idptr = calloc (1, sizeof (*idptr)))
+        || flux_future_aux_set (f, "jobid", idptr, free) < 0) {
+        flux_kvs_txn_destroy (txn);
+        flux_future_destroy (f);
+        free (idptr);
+        return -1;
+    }
+    *idptr = id;
+    flux_kvs_txn_destroy (txn);
+    return 0;
+}
+
+
+static int sched_cb (flux_plugin_t *p,
+                     const char *topic,
+                     flux_plugin_arg_t *args,
+                     void *arg)
+{
+    const char *R = NULL, *jobspec = NULL;
+    flux_jobid_t id;
+    flux_future_t *f;
+    int retval = -1;
+
+    /*  If flux-kube::R set on this job then commit R to KVS,
+     *  get the jobspec, submit the jobspec to flux-kube.py, 
+     *  and set flux-kube flag.
+     */
+    if (!(R = flux_jobtap_job_aux_get (p,
+                                       FLUX_JOBTAP_CURRENT_JOB,
+                                       "flux-kube::R")))
+        return 0;
+
+    if (!(jobspec = flux_jobtap_job_aux_get (p,
+                                             FLUX_JOBTAP_CURRENT_JOB,
+                                             "flux-kube::jobspec")))
+        return 0;
```

Comment:
Suggestion: If `flux-kube::R` exists for a job, but `flux-kube::jobspec` does not, should that result in a job exception? Currently that condition is ignored (which may be fine, but I thought I'd ask)

Reply:
Good catch; I need to fix this.